### PR TITLE
DMP-2767: Auditing admin courthouse operations

### PR DESCRIPTION
**Pull request by tonymort**
https://tools.hmcts.net/jira/browse/DMP-2767

Auditing admin courthouse operations

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/courthouse/service/CourthouseServiceImpl.java [hunk 1]
```diff
@@ -285,11 +297,66 @@ public AdminCourthouse updateCourthouse(Integer courthouseId, CourthousePatch co
...
+
+    private static boolean securityGroupsUpdated(CourthousePatch courthousePatch, CourthouseEntity prePatchedEntity) {
+        if (courthousePatch.getSecurityGroupIds() == null) {
+            return false;
+        }
+        var prePatchedGroups = prePatchedEntity.getSecurityGroups().stream()
+            .map(SecurityGroupEntity::getId)
+            .toList();
+        return !prePatchedGroups.equals(courthousePatch.getSecurityGroupIds());
+    }
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/darts/courthouse/service/CourthouseServiceImpl.java:361]: I might be missing something but I wonder whether we need to write this custom audit logic. 
The power of Envers is that it provides automatic auditing by just adding annotations. I would have thought that we could cover the basic admin court case operations requirements we have by using only annotations. 
And then use the Envers API to figure out what type of changes where made to the entity when the need arise to access those changes.

Comment (tonymort) [reply src/main/java/uk/gov/hmcts/darts/courthouse/service/CourthouseServiceImpl.java:361]: This isn't related to envers, we have custom auditing also.

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -86,7 +86,7 @@ spring:
     properties:
       hibernate:
         default_schema: ${spring.datasource.schema}
-        dialect: org.hibernate.dialect.PostgreSQLDialect
+#        dialect: org.hibernate.dialect.PostgreSQLDialect
```
[the remark this answers is not in the document]
Comment (tonymort) [reply src/main/resources/application.yaml:89]: Will uncomment, according to documentation we shouldn't need this however, hibernate 6+ figures it out from the JDBC url.  We know we want PostgreSQLDialect so won't do any harm

File: src/main/java/uk/gov/hmcts/darts/courthouse/service/CourthouseServiceImpl.java (hunk 1 above)
Comment (jackmaloney) [reply src/main/java/uk/gov/hmcts/darts/courthouse/service/CourthouseServiceImpl.java:361]: I believe the requirement is to record that something happened in the 'audit' table and then if they want more detail about what actually changed, they can query the envers table.

**Review by mario-paniccia — APPROVED** [review]

**Pull request merged**